test/asset-swapper/exchange_proxy_swap_quote_consumer_test.ts

```diff
@@ -509,5 +601,69 @@ describe('ExchangeProxySwapQuoteConsumer', () => {
             expect(payTakerTransformerData.amounts).to.deep.eq([]);
             expect(payTakerTransformerData.tokens).to.deep.eq([TAKER_TOKEN, ETH_TOKEN_ADDRESS]);
         });
+
+        it('Uses two `FillQuoteTransformer`s if given a mixed two-hop (Otc, Limit) sell quote', async () => {
+            const quote = getRandomTwoHopQuote(MarketOperation.Sell, [
+                FillQuoteTransformerOrderType.Otc,
+                FillQuoteTransformerOrderType.Limit,
+            ]) as MarketSellSwapQuote;
+            const callInfo = await consumer.getCalldataOrThrowAsync(quote, {
+                extensionContractOpts: { isTwoHop: true },
+            });
+            const callArgs = transformERC20Encoder.decode(callInfo.calldataHexString) as TransformERC20Args;
+            expect(callArgs.inputToken).to.eq(TAKER_TOKEN);
+            expect(callArgs.outputToken).to.eq(MAKER_TOKEN);
+            expect(callArgs.inputTokenAmount).to.bignumber.eq(quote.worstCaseQuoteInfo.totalTakerAmount);
+            expect(callArgs.minOutputTokenAmount).to.bignumber.eq(quote.worstCaseQuoteInfo.makerAmount);
+            expect(callArgs.transformations).to.be.length(3);
+            expect(
+                callArgs.transformations[0].deploymentNonce.toNumber() ===
+                    consumer.transformerNonces.fillQuoteTransformer,
+            );
+            expect(
+                callArgs.transformations[1].deploymentNonce.toNumber() ===
+                    consumer.transformerNonces.fillQuoteTransformer,
+            );
+            expect(
+                callArgs.transformations[2].deploymentNonce.toNumber() ===
+                    consumer.transformerNonces.payTakerTransformer,
+            );
+            const [firstHopOrder, secondHopOrder] = quote.orders;
+            const firstHopFillQuoteTransformerData = decodeFillQuoteTransformerData(callArgs.transformations[0].data);
```

Comment:
optional: if this is series of `expect` is used a lot maybe it could be factor out into a helper function.